src/main/scala/com/fortysevendeg/scala/android/modules/utils/validation.scala

```diff
@@ -0,0 +1,45 @@
+package com.fortysevendeg.scala.android.modules.utils
+
+import scalaz._
+import scalaz.std.list._
+import scalaz.syntax.traverse._
+
+object validation {
+
+  val defaultMessage = "Unexpected validation error."
+
+  type Validator[T] = (T) => Validation[ValidationMessage, T]
+
+  case class ValidationMessage(
+      message: String,
+      details: String = "")
+
+  case class ValidationException(
+      errorHead: ValidationMessage,
+      errorTail: Seq[ValidationMessage] = Seq.empty)
+      extends RuntimeException(s"${errorHead.message} - ${errorHead.details}")
+
+  def WithValidation[T, ValT <: T, A, S]
+  (validators: ((ValT) => Validation[ValidationMessage, T])*)
+      (happyPath: => S)
+      (implicit input: ValT, m: Manifest[T]): S = {
+
+    validators.toList.map(p => p(input).toValidationNel).sequenceU match {
+      case Failure(NonEmptyList(head: ValidationMessage)) => throw ValidationException(head)
+      case Failure(list: NonEmptyList[_]) =>
+        throw ValidationException(list.head.asInstanceOf[ValidationMessage], list.tail.asInstanceOf[List[ValidationMessage]])
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of head/last on collections.](https://www.codacy.com/app/47Degrees/scala-android/file/661377143/issues/source?bid=2339068&fileBranchId=2339108#l30)